packages/compiler/src/server/tspconfig/completion.ts

```diff
@@ -80,6 +124,99 @@ export async function provideTspconfigCompletionItems(
         itemsFromEmitter.push(...more);
       }
       return [...itemsFromBuiltIn, ...itemsFromEmitter];
+    } else if (nodePath.length > CONFIG_PATH_LENGTH_FOR_LINTER_LIST && nodePath[0] === "linter") {
+      const extendKeyWord = nodePath[CONFIG_PATH_LENGTH_FOR_LINTER_LIST - 1];
+      const items: CompletionItem[] = [];
+      const linters = await linterProvider.listLibraries(tspConfigFile);
+
+      if (extendKeyWord === "extends") {
+        for (const [name, pkg] of Object.entries(linters)) {
+          // If a ruleSet exists for the linter, add it to the end of the library name.
+          const exports = await pkg.getModuleExports();
+          if (exports?.$linter?.ruleSets !== undefined) {
+            // Below ruleSets are objects rather than arrays
+            for (const [ruleSet] of Object.entries(exports?.$linter?.ruleSets)) {
```

Comment:
I remember Object.entries will throw if undefined is given. please help to confirm. same comment for other places using Object.entries.

Reply:
Yes, Object.entries will throw, but judgment has ben made before use

Reply:
then please remove the ? so that compiler can help you to catch it in case the pre-check is removed for some reason in the future.